database/migrations/2019_10_24_183410_add_polymorphic_relationship_to_degrees.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class AddPolymorphicRelationshipToDegrees extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('degrees', function (Blueprint $table) {
+            $table->dropForeign(['applicant_id']);
+
+            $table->renameColumn('applicant_id', 'degreeable_id');
+            $table->string('degreeable_type')->default('applicant'); // Set default for existing rows.
```

Comment:
Hmm. Maybe the laravel migration `->default` method is doing some extra work then, because after running these migrations existing rows definitely have the 'applicant' type.

Reply:
You are correct, just tested and it works. Not sure how, based on my reading but it's good to know.